pallet-chainlink-feed/src/benchmarking.rs

```diff
@@ -0,0 +1,333 @@
+	verify {
+		assert_last_event::<T>(Event::Frozen(Default::default(), caller).into());
+	}
+
+	thaw {
+		let (caller, caller_lookup) = create_default_minted_asset::<T>(10, 100u32.into());
+		Assets::<T>::freeze(
+			SystemOrigin::Signed(caller.clone()).into(),
+			Default::default(),
+			caller_lookup.clone(),
+		)?;
+	}: _(SystemOrigin::Signed(caller.clone()), Default::default(), caller_lookup)
+	verify {
+		assert_last_event::<T>(Event::Thawed(Default::default(), caller).into());
+	}
+
+	freeze_asset {
+		let (caller, caller_lookup) = create_default_minted_asset::<T>(10, 100u32.into());
+	}: _(SystemOrigin::Signed(caller.clone()), Default::default())
+	verify {
+		assert_last_event::<T>(Event::AssetFrozen(Default::default()).into());
+	}
+
+	thaw_asset {
+		let (caller, caller_lookup) = create_default_minted_asset::<T>(10, 100u32.into());
+		Assets::<T>::freeze_asset(
+			SystemOrigin::Signed(caller.clone()).into(),
+			Default::default(),
+		)?;
+	}: _(SystemOrigin::Signed(caller.clone()), Default::default())
+	verify {
+		assert_last_event::<T>(Event::AssetThawed(Default::default()).into());
+	}
+
+	transfer_ownership {
+		let (caller, _) = create_default_asset::<T>(10);
+		let target: T::AccountId = account("target", 0, SEED);
+		let target_lookup = T::Lookup::unlookup(target.clone());
+	}: _(SystemOrigin::Signed(caller), Default::default(), target_lookup)
+	verify {
+		assert_last_event::<T>(Event::OwnerChanged(Default::default(), target).into());
+	}
+
+	set_team {
+		let (caller, _) = create_default_asset::<T>(10);
+		let target0 = T::Lookup::unlookup(account("target", 0, SEED));
+		let target1 = T::Lookup::unlookup(account("target", 1, SEED));
+		let target2 = T::Lookup::unlookup(account("target", 2, SEED));
+	}: _(SystemOrigin::Signed(caller), Default::default(), target0.clone(), target1.clone(), target2.clone())
+	verify {
+		assert_last_event::<T>(Event::TeamChanged(
+			Default::default(),
+			account("target", 0, SEED),
+			account("target", 1, SEED),
+			account("target", 2, SEED),
+		).into());
+	}
+
+	set_max_zombies {
+		let (caller, _) = create_default_asset::<T>(10);
+		let max_zombies: u32 = 100;
+		T::Currency::make_free_balance_be(&caller, BalanceOf::<T>::max_value());
+	}: _(SystemOrigin::Signed(caller), Default::default(), max_zombies)
+	verify {
+		assert_last_event::<T>(Event::MaxZombiesChanged(Default::default(), max_zombies).into());
+	}
+
+	set_metadata {
+		let n in 0 .. T::StringLimit::get();
+		let s in 0 .. T::StringLimit::get();
+
+		let name = vec![0u8; n as usize];
+		let symbol = vec![0u8; s as usize];
+		let decimals = 12;
+
+		let (caller, _) = create_default_asset::<T>(10);
+		T::Currency::make_free_balance_be(&caller, BalanceOf::<T>::max_value());
+	}: _(SystemOrigin::Signed(caller), Default::default(), name.clone(), symbol.clone(), decimals)
+	verify {
+		assert_last_event::<T>(Event::MetadataSet(Default::default(), name, symbol, decimals).into());
+	}
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
we have a macro for this now.

Reply:
but is it in Substrate v2?

Reply:
no :(

Reply:
why not move to 3.0.0 😈

Reply:
cannot at this moment because the client libraries are in go which are not updated to v3, yet